pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -465,7 +554,40 @@ abstract contract BasePool is IBasePool, BasePoolAuthorization, BalancerPoolToke
         bytes memory userData
     ) internal virtual returns (uint256 bptAmountIn, uint256[] memory amountsOut);
 
-    // Internal functions
+    /**
+     * @dev A minimal proportional exit, suitable as is for most pools: though not for pools with Phantom BPT
+     * or other special considerations. Designed to be overridden if a pool needs to do extra processing,
+     * such as scaling a stored invariant, or caching the new total supply.
```

Comment:
An override would be a bit tricky as we'd likely want access to the BPT ratio. But we can look into that later.